shopfloor/services/location_content_transfer.py

```diff
@@ -252,24 +235,16 @@ def _create_moves_from_location(self, location):
 
     def _find_location_to_work_from(self):
         location = self.env["stock.location"]
-        pickings = self.env["stock.picking"].search(
-            [
-                ("picking_type_id", "in", self.picking_types.ids),
-                ("state", "=", "assigned"),
-                ("user_id", "in", (False, self.env.user.id)),
-            ],
-            order="user_id, priority desc, scheduled_date asc, id desc",
-        )
-
-        for next_picking in pickings:
-            move_lines = next_picking.move_line_ids.filtered(
-                lambda line: line.qty_done < line.reserved_uom_qty
-            )
-            location = fields.first(move_lines).location_id
-            if location:
-                break
+        move_lines = self.search_move_line.search_move_lines(match_user=True)
+        for move_line in move_lines:
+            if move_line.location_id:
+                return move_line.location_id
         return location
```

Comment:
`location_id` is a required field. You can just do
```suggestion
        return fields.first(move_lines).location_id
```